File: 0242-valid-anagram/0242-valid-anagram.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
        return false
    }
    freq := [26]int{}

    for _, char := range s {
        freq[char-'a']++
    }

    for _, char := range t {
        freq[char-'a']--
        if freq[char-'a'] < 0 {
            return false
        }
    }

    return true
}

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        char[] sCharCount = new char[26];
        char[] tCharCount = new char[26];
        
        for (char c : s.toCharArray()){
            sCharCount[c -'a'] += 1;
        }

        for (char c : t.toCharArray()){
            tCharCount[c -'a'] += 1;
        }

        return Arrays.equals(sCharCount, tCharCount);
    }
}